pkg/volume/util/subpath/subpath_linux.go

```diff
@@ -240,7 +240,7 @@ func doCleanSubPaths(mounter mount.Interface, podDir string, volumeName string)
 				return err
 			}
 
-			if info.IsDir() {
+			if info != nil && info.IsDir() {
```

Comment:
Do you know when info is nil?

Reply:
@msau42 
From the kubelet log, it seems the info was nil during volume unmounting.

Reply:
Yes but why? File got deleted? Permission error?

Reply:
The file was deleted by the time this code was executed.

Reply:
should we be checking the incoming `err` parameter instead of this (that's when `info` is nil)

Reply:
see https://golang.org/pkg/path/filepath/#WalkFunc

Reply:
@liggitt 
Your comment has been addressed.

Reply:
I'm trying to understand the original intent of this comment:

```
// pass through errors and let doCleanSubPath handle them
```

that method intentionally handles not found errors and returns nil...

Reply:
The error return from doCleanSubPath() overwrites the incoming error.

Reply:
This function was changed to `filepath.Walk` in https://github.com/kubernetes/kubernetes/pull/71804 in order to handle teardown of stale/corrupt nfs mounts. That is why we pass through the error.

In the case of notfound (and info == nil), I think skipping and continuing is the right thing to do

Reply:
https://github.com/kubernetes/kubernetes/pull/82698 is what introduced the panic in 1.17. So we should backport the fix as well

Reply:
in that case, the original `if info != nil && info.IsDir()` approach might be simplest, but this needs a comment explaining when info is nil

Reply:
I guess what I'm still wondering though, is what else is deleting the files at this time?  This is not supposed to run into all the pod's containers have already been terminated, so the only thing running should be volume cleanup, and that is supposed to be serialized per volume

Reply:
I have reverted to the initial fix and added a comment above the check.

Reply:
I'm going to debug https://github.com/kubernetes/kubernetes/pull/86277#discussion_r358508595 further to figure out what's going on

Reply:
Ok I figured out that cause. It is indeed when you have stale nfs mounts, not file doesn't exist, triggered by this test case: [driver: nfs] subPath should be able to unmount after the subpath directory is deleted.

The reason why the test didn't catch the issue is because `doCleanSubPath` successfully unmounts, it's the processing afterwards that panics the goroutine, but the retry will resolve it.